deps/rabbitmq_stream/src/rabbit_stream_reader.erl

```diff
@@ -3265,6 +3245,34 @@ ensure_outstanding_requests_timer(#stream_connection{
 ensure_outstanding_requests_timer(C) ->
     C.
 
+ensure_token_expiry_timer(User, #stream_connection{token_expiry_timer = Timer} = Conn) ->
+    TimerRef =
+    maybe
+        rabbit_log:debug("Checking token expiry"),
+        true ?= rabbit_access_control:permission_cache_can_expire(User),
```

Comment:
I wonder whether we can remove function `rabbit_access_control:permission_cache_can_expire/1` in future? This function seems to be unnecessary given that we now have `rabbit_access_control:expiry_timestamp/1` which will return `never` if the permission cache cannot expire. Ideally `rabbit_channel` will use the same new approach as done by Stream connections in this PR. This change doesn't have to be part of this PR though. However, ideally we don't add new calls to `rabbit_access_control:permission_cache_can_expire/1` such that we can remove that funtion more easily in the future. WDYT?

Reply:
Sounds reasonable. Do we try to remove `rabbit_access_control:permission_cache_can_expire/1` in this PR? The sooner, the better.

Reply:
It's up to you :) I also think, the sooner the better, yes.

Reply:
https://github.com/rabbitmq/rabbitmq-server/issues/10382